Doc/howto/clinic.rst

```diff
@@ -93,67 +96,148 @@ and it should be able to do many interesting and smart
+All of these lines, including the :term:`start line` and :term:`checksum line`,
+are collectively called an Argument Clinic :term:`block`:
 
 .. code-block:: none
 
     /*[clinic input]
     ... clinic input goes here ...
     [clinic start generated code]*/
     ... clinic output goes here ...
-    /*[clinic end generated code: checksum=...]*/
+    /*[clinic end generated code: ...]*/
 
 If you run Argument Clinic on the same file a second time, Argument Clinic
-will discard the old output and write out the new output with a fresh checksum
-line.  However, if the input hasn't changed, the output won't change either.
-
-You should never modify the output portion of an Argument Clinic block.  Instead,
-change the input until it produces the output you want.  (That's the purpose of the
-checksum—to detect if someone changed the output, as these edits would be lost
-the next time Argument Clinic writes out fresh output.)
-
-For the sake of clarity, here's the terminology we'll use with Argument Clinic:
-
-* The first line of the comment (``/*[clinic input]``) is the *start line*.
-* The last line of the initial comment (``[clinic start generated code]*/``) is the *end line*.
-* The last line (``/*[clinic end generated code: checksum=...]*/``) is the *checksum line*.
-* In between the start line and the end line is the *input*.
-* In between the end line and the checksum line is the *output*.
-* All the text collectively, from the start line to the checksum line inclusively,
-  is the *block*.  (A block that hasn't been successfully processed by Argument
-  Clinic yet doesn't have output or a checksum line, but it's still considered
-  a block.)
+will discard the old :term:`output` and write out the new output with a fresh
+:term:`checksum line`.
+If the :term:`input` hasn't changed, the output won't change either.
+
+.. note::
+
+   You should never modify the output of an Argument Clinic block,
+   as any change will be lost in future Argument Clinic runs;
+   Argument Clinic will detect an output checksum mismatch and regenerate the
+   correct output.
+   If you are not happy with the generated output,
+   you should instead change the input until it produces the output you want.
+
+
+.. _clinic-reference:
+
+Reference
+=========
+
+
+.. _clinic-terminology:
+
+Terminology
+-----------
+
+.. glossary::
+
+  start line
+     The line ``/*[clinic input]``.
+     This line marks the beginning of Argument Clinic input.
+     Notice that the *start line* opens a C block comment.
+
+  end line
+     The line ``[clinic start generated code]*/``.
+     This line marks the end of Argument Clinic input.
+     Notice that the *end line* closes the C block comment opened
+     by the *start line*.
+
+  checksum line
+     A line that looks like ``/*[clinic end generated code: ...]*/``.
+     The three dots will be replaced by the actual checksum(s).
+     The checksum line marks the end of Argument Clinic generated code.
```

Comment:
Could you link to a definition somewhere else of what "checksum" means? Feels like we're maybe assuming a pre-existing knowledge of some jargon here

Reply:
Good point!

Reply:
See 9bdd1c5fc0551555d6c883b6ad386b5e9bd7b9e9

Reply:
Continued in https://github.com/python/cpython/pull/106981#discussion_r1274841142